digdag-standards/src/main/java/io/digdag/standards/operator/ForRangeOperatorFactory.java

```diff
@@ -0,0 +1,113 @@
+    {
+        return new ForRangeOperator(context);
+    }
+
+    static class ForRangeOperator
+            implements Operator
+    {
+        private final TaskRequest request;
+
+        public ForRangeOperator(OperatorContext context)
+        {
+            this.request = context.getTaskRequest();
+        }
+
+        @Override
+        public TaskResult run()
+        {
+            Config params = request.getConfig();
+
+            Config doConfig = request.getConfig().getNested("_do");
+            boolean parallel = params.get("_parallel", boolean.class, false);
+
+            Config rangeConfig = params.parseNested("_command");
+            long from = rangeConfig.get("from", long.class);
+            long until = rangeConfig.get("until", long.class);
+
+            Optional<Long> stepConfig = request.getConfig().getOptional("step", Long.class);
+            Optional<Long> countConfig = request.getConfig().getOptional("count", Long.class);
+
+            if (stepConfig.isPresent() && countConfig.isPresent()) {
+                throw new ConfigException("Setting both step and count options to for_range is invalid");
+            }
+            if (!stepConfig.isPresent() && !countConfig.isPresent()) {
+                throw new ConfigException("step or count option is required for for_range");
+            }
+
+            long step = stepConfig.or(() -> (until - from + (countConfig.get() - 1)) / countConfig.get());
+
+            int count = 0;
+            Config generated = doConfig.getFactory().create();
+            for (long pos = from; pos < until; pos += step) {
+                long rangeFrom = pos;
+                long rangeUntil = Math.min(pos + step, until);
+                Config exportParams = params.getFactory().create();
+                exportParams.getNestedOrSetEmpty("range")
+                    .set("from", rangeFrom)
+                    .set("until", rangeUntil);
+                Config subtask = params.getFactory().create();
+                subtask.setAll(doConfig);
+                subtask.getNestedOrSetEmpty("_export").setAll(exportParams);
+                generated.set(
+                        buildTaskName(rangeFrom, rangeUntil),
+                        subtask);
+                count++;
+            }
+
+            enforceTaskCountLimit(count);
+
+            if (parallel) {
+                generated.set("_parallel", parallel);
+            }
+
+            return TaskResult.defaultBuilder(request)
+                .subtaskConfig(generated)
+                .build();
+        }
+
+        private static String buildTaskName(long rangeFrom, long rangeUntil)
+        {
+            return String.format("+range-from=%d&until=%d", rangeFrom, rangeUntil);
+        }
+
+        private static void enforceTaskCountLimit(int size)
+        {
+            if (size > Limits.maxWorkflowTasks()) {
+                throw new ConfigException("Too many for_range subtasks. Limit: " + Limits.maxWorkflowTasks());
```

Comment:
Showing `size` here would help users to understand what's wrong since it's likely users don't know an actual number of tasks with `step`.

Reply:
That's very good point.